apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -108,7 +108,7 @@ defmodule OMG.Watcher.BlockGetter do
   end
 
   def handle_continue({:apply_block_step, :execute_transactions, block_application}, state) do
-    tx_exec_results = for(tx <- block_application.transactions, do: OMG.State.exec(tx, :ignore))
+    tx_exec_results = for(tx <- block_application.transactions, do: OMG.State.exec(tx, :no_fees_transaction))
```

Comment:
so I'd say `no_fees_transaction` is slightly misleading. It makes sense from `MergeTransactionValidators` PoV, but doesn't from `OMG.State.exec` PoV. `exec` doesn't care about the reason, it's just being told to `ignore` fees.

Moreover, it still may be, that the merge transaction will pay fees (b/c someone didn't use the awesome merge tx discount).

if `:ignore` is bothering you, maybe `:ignore_fees` or `:ignore_fee_check` or `:no_fees_required`.

And going back to the question from the PR description, about why we do this in such 2 phases: we want to keep whatever `OMG.State` process is doing as lightweight as possible. So we figure out how much the tx should pay entering the context of `OMG.State.exec`. Also the latter part is used in the Watcher as well, and the Watcher `ignores` the fees altogether :)

Reply:
@pdobacz Fair enough! I like `:no_fees_required` the most, what do you guys think @InoMurko @achiurizo?

Thanks for the explanation on the 2 phases :)

Reply:
> Thanks for the explanation on the 2 phases :)

whoops, I just noticed I actually missed a word that changes the sense of the sentence there :joy: . Fixed now

Reply:
In this case, fees are not applied because Watcher is simply processing what has already been spent... so maybe we should name that accordingly.`:watcher_zero_fee` or something like that? Though `:no_fees_required` is good for me as well.

Reply:
Also :+1: on `:no_fees_required`, *much* more clearer what the intention is here.